src/main/java/com/pusher/rest/Pusher.java

```diff
@@ -341,9 +372,14 @@ public Result trigger(final List<String> channels, final String eventName, final
         Prerequisites.noNullMembers("channels", channels);
         Prerequisites.areValidChannels(channels);
         Prerequisites.isValidSocketId(socketId);
-
-        final String body = BODY_SERIALISER.toJson(new TriggerData(channels, eventName, serialise(data), socketId));
-
+        Prerequisites.eitherOneorNoEncryptedChannels(channels);
+        String body;
+        if(Crypto.isEncryptedChannel(channels.get(0))) {
+            EncryptedPayload dataEncrypted = pusherCrypto.encrypt(channels.get(0), data);
+            body = BODY_SERIALISER.toJson(new TriggerData(channels, eventName, serialise(dataEncrypted), socketId));
+        } else {
+            body = BODY_SERIALISER.toJson(new TriggerData(channels, eventName, serialise(data), socketId));
+        }
```

Comment:
This PR is missing encryption of events in batch triggers.

Reply:
Was not aware we were steaming ahead with it. Added now!